lib/util/search/address_suggestion_manager.dart

```diff
@@ -97,29 +101,29 @@ class AddressSuggestionManager {
   //     'Content-Type': 'application/json',
   //   },
   // );
-  Future<List<dynamic>> _doRequest(
+  Future<List<Map<String, dynamic>>> _doRequest(
     String elm,
     String target,
     String contentType,
     String query,
   ) async {
-    if (query.length < searchLengthRequirement) return [];
+    if (query.length < searchLengthRequirement) return <Map<String, dynamic>>[];
 
     HttpClientRequest request = await _httpClient.postUrl(
       Uri(
         scheme: 'https',
         host: _motisHost,
-        queryParameters: {'elm': elm},
+        queryParameters: <String, String>{'elm': elm},
       ),
     );
 
-    request.add(utf8.encode(json.encode({
-      "destination": {"type": "Module", "target": target},
+    request.add(utf8.encode(json.encode(<String, dynamic>{
+      "destination": <String, dynamic>{"type": "Module", "target": target},
       "content_type": contentType,
-      "content": {"input": query}
+      "content": <String, dynamic>{"input": query}
     })));
 
-    String response = await request.close().then((value) => value.transform(utf8.decoder).join());
+    String response = await request.close().then((HttpClientResponse value) => value.transform(utf8.decoder).join());
 
     Map<String, dynamic> responseMap = json.decode(response);
     Map<String, dynamic> content = responseMap['content'];
```

Comment:
line 131 is the same Exception thrown when i hit enter and the AddresSuggestions are not loaded

Reply:
I think I fixed that? Can you verify? I added a new package that helps with picking those deeply nested values. But I have no clue why it does not work with normal dart code... very confusing